### ISSUE TITLE
[Bug]: All electron application have frame thickness mismatch (small pixel borders around window title) under Linux with Wayland

### ISSUE THREAD

**Issue by cedricroijakkers** [electron/electron#42234]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

30.0.6

### What operating system are you using?

Other Linux

### Operating System Version

Linux 48HVWD3 6.9.1-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 17 May 2024 16:56:38 +0000 x86_64 GNU/Linux

### What arch are you using?

x64

### Last Known Working Electron version

None, never seen this working under Wayland

### Expected Behavior

When using any Electron app under Linux with Wayand and setting the window to full screen, the window title, just like the rest of the window, should fill the entire screen.

### Actual Behavior

The window title is missing a few pixels on the left, top, and right border of the title only.

I can reproduce this issue under Linux with Wayland and GNOME 46, with 125% scaling on my screen.

To reproduce:

- Open a terminal
- Run electron30 with the Wayland flags: `electron30 --enable-features=WaylandWindowDecorations --ozone-platform-hint=auto`
- Maximize the electron window that appears

The terminal will show an error:

```
[179576:0521/164350.919483:ERROR:window_frame_provider_gtk.cc(407)] Frame thickness mismatch!  Old: [x:5,5 y:2,8], new: [x:5,5 y:3,8].  Current scale is 1.25.  Please report to crbug.com/1240905.
```

If you look closely at the title bar of the application, you will see the desktop background around the top, left, and right of the title bar only. The rest of the window is correctly maximized.

See screenshot:

![image](https://github.com/electron/electron/assets/15158042/9d808eaa-af6f-48e9-81aa-5ee3066979d5)

**Comment by ZtereoHYPE** [comment]
Noticed the same issue. Here are some attachments, both with and without fractional scaling:
![image](https://github.com/electron/electron/assets/57519662/b6216d4d-d878-45e2-87f8-2303f0666021)
![image](https://github.com/electron/electron/assets/57519662/f235dfe5-2434-4aa8-938d-4d07400f3e13)

**Comment by cedricroijakkers** [comment]
Still can reproduce this in 30.0.9

**Comment by cedricroijakkers** [comment]
Still can reproduce this in 30.1.0

**Comment by selfisekai** [comment]
can reproduce, seems specific to GNOME? can't reproduce on KDE

**Comment by ZtereoHYPE** [comment]
To add some information that might be helpful, it seems like some surfaces are being rendered with an offset of one pixel in both axis, which is visible by zooming into the screenshots above. 

![Screenshot_20240609-011946](https://github.com/electron/electron/assets/57519662/fab25d3a-d6bd-422c-894b-3fa6ad6eadd4)

Could this be a rounding or starting from 1 rather than 0 type of bug, maybe related to the GTK theme, considering it seems to only happen in GNOME?

**Comment by Raindrac** [comment]
Not stale.

**Comment by cedricroijakkers** [comment]
I can still reproduce this issue in Electron 32.1.0

**Comment by cedricroijakkers** [comment]
Can stil reproduce in Electron 32.2.7.

**Comment by glyh** [comment]
Yeah, I can confirm I can still reproduce this.

**Comment by 01111010t** [comment]
It appears that I'm experiencing this bug with electron 35.1 on wayland:

![Image](https://github.com/user-attachments/assets/e8f013c4-8116-476f-a925-eeee7be07220)

**Comment by cedricroijakkers** [comment]
Still can reproduce this in 33.4.

**Comment by cedricroijakkers** [comment]
Guess what, this issue is not magically solved, so: bump.

**Comment by ZtereoHYPE** [comment]
Doing some "research" it appears that this is indeed tied to the GTK theme in Chrome/Chromium, as it happens in the browser too while said theme is active. Since the electron folks have shown zero interest in this, perhaps it might be worth opening an issue on Chromium’s tracker?

**Comment by codebytere** [comment]
yes - this is something we inherit and don't have control over directly without patching which we strongly prefer to avoid when there are alternatives. This should be opened as an issue on Chromium's [tracker](https://issues.chromium.org/issues) and we'll inherit a fix when one is made.

**Comment by ZtereoHYPE** [comment]
For future reference, this issue seems to match the closest:
https://issues.chromium.org/issues/40900734

Unfortunately, like many issues related to fractional scaling, it has been open for over 2 years with no action :/
